service/invoker/invoker.go

```diff
@@ -55,17 +47,57 @@ func New(index archive.Reader, options ...func(*Config)) (*Invoker, error) {
 		return nil, fmt.Errorf("could not initialize cache: %w", err)
 	}
 
-	i := Invoker{
-		index: index,
-		vm:    vm,
-		cache: cache,
-	}
-
-	vmCtx := fvm.NewContext(fvm.WithBlocks(&i))
-
-	i.vmCtx = vmCtx
-
-	return &i, nil
+	blocks := NewBlocks(index)
+
+	// This is copied code from flow-go engine/execution/computation/manager.go
+	// Ideally the same code would be used, but that requires flow-go changes.
+	// TODO: reuse code from flow-go
+	var vm fvm.VM
+	if cfg.NewCustomVirtualMachine != nil {
+		vm = cfg.NewCustomVirtualMachine()
+	} else {
+		vm = fvm.NewVirtualMachine()
+	}
+
+	chainID := cfg.ChainID
+	fvmOptions := []fvm.Option{
+		fvm.WithReusableCadenceRuntimePool(
+			reusableRuntime.NewReusableCadenceRuntimePool(
+				computation.ReusableCadenceRuntimePoolSize,
+				runtime.Config{
+					TracingEnabled:        cfg.CadenceTracing,
+					AccountLinkingEnabled: true,
+					// Attachments are enabled everywhere except for Mainnet
+					AttachmentsEnabled: chainID != flow.Mainnet,
+					// Capability Controllers are enabled everywhere except for Mainnet
+					CapabilityControllersEnabled: chainID != flow.Mainnet,
```

Comment:
should we leave a `todo` here for when mainnet has these? I'm assuming they'll appear after the HCU?

Reply:
I added a comment above, saying where this is copied from. When that changes, this should change too. Ideally we would use the same code for both, and that will be the next step.